pyflakes/checker.py

```diff
@@ -518,6 +527,60 @@ def is_typing_overload_decorator(node):
     )
 
 
+def make_tokens(code):
+    # PY3: tokenize.tokenize requires readline of bytes
+    if not isinstance(code, bytes):
+        code = code.encode('UTF-8')
+    lines = iter(code.splitlines(True))
+    # next(lines, b'') is to prevent an error in pypy3
+    return tuple(tokenize_tokenize(lambda: next(lines, b'')))
+
+
+class _TypeableVisitor(ast.NodeVisitor):
+    """Collect the line number and nodes which are deemed typeable by
+    PEP 484
+
+    https://www.python.org/dev/peps/pep-0484/#type-comments
+    """
+    def __init__(self):
+        self.typeable_lines = []  # type: int
+        self.typeable_nodes = {}  # type: Dict[int, ast.AST]
+
+    def _typeable(self, node):
+        # if there is more than one typeable thing on a line last one wins
+        self.typeable_lines.append(node.lineno)
```

Comment:
Perhaps this is silly, but why are we doubly tracking the line number? Could `typeable_lines` simply be a property that does `list(self.typeable_nodes.values())`?

Reply:
I couldn't think of a way to make the lookup efficient without needing two state variables here.  having a property which sorts the keys would invoke a O(N) (`list(typeable_nodes)`) + O(N*log(N)) (`sorted(...)`) + O(log(N)) (`bisect.bisect_right(...)`) instead of just the `O(log(N))` lookup

Reply:
Makes sense. :+1: